File: ShopAplication/app/src/main/java/com/example/shopaplication/RegestairActivity.java
package com.example.shopaplication;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.app.ProgressDialog;
import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.database.ValueEventListener;

import java.util.HashMap;

public class RegestairActivity extends AppCompatActivity {
  private EditText edit_name,edit_pass,edit_number;
  private Button btn_register;
  private ProgressDialog progressDialog;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_regestair);
        edit_name=findViewById(R.id.edit_textname_register);
        edit_number=findViewById(R.id.edit_textnumper_register);
        edit_pass=findViewById(R.id.edit_textpass_register);
        btn_register=findViewById(R.id.btn_register);
        progressDialog=new ProgressDialog(this);


        btn_register.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                CreatAccount();

            }
        });

    }
    private void CreatAccount(){
        String name=edit_name.getText().toString();
        String number=edit_number.getText().toString();
        String pass=edit_pass.getText().toString();

        if(TextUtils.isEmpty(name)){
            Toast.makeText(this, "Please Enter Name", Toast.LENGTH_SHORT).show();
        }else if(TextUtils.isEmpty(number)){
            Toast.makeText(this, "Please Enter Number", Toast.LENGTH_SHORT).show();

        }else if(TextUtils.isEmpty(pass)){
            Toast.makeText(this, "Please Enter passwarad", Toast.LENGTH_SHORT).show();
        }else {
            progressDialog.setTitle("Create Account");
            progressDialog.setMessage("please Wait........");
            progressDialog.setCanceledOnTouchOutside(false);
            progressDialog.show();
            CreatAccountFirebase(name,number,pass);
        }

    }

    private void CreatAccountFirebase(String name, String number, String pass) {
        final DatabaseReference ref;
        ref= FirebaseDatabase.getInstance().getReference();
        ref.addListenerForSingleValueEvent(new ValueEventListener() {
            @Override
            public void onDataChange(@NonNull DataSnapshot snapshot) {
                if ((snapshot.child("users").child(number).exists())) {

                    Toast.makeText(RegestairActivity.this, "Number is exist", Toast.LENGTH_SHORT).show();
                    Intent intent = new Intent(RegestairActivity.this, MainActivity.class);
                    startActivity(intent);
                } else {
                    HashMap<String, Object> map = new HashMap<>();
                    map.put("name", name);
                    map.put("numper", number);
                    map.put("pass", pass);
                    ref.child("users").child(number).updateChildren(map).addOnCompleteListener(new OnCompleteListener<Void>() {
                        @Override
                        public void onComplete(@NonNull Task<Void> task) {
                            progressDialog.dismiss();
                            Toast.makeText(RegestairActivity.this, "Create Account Sucess", Toast.LENGTH_SHORT).show();
                            Intent intent = new Intent(RegestairActivity.this, MainActivity.class);
                            startActivity(intent);

                        }
                    });
                }
            }



            @Override
            public void onCancelled(@NonNull DatabaseError error) {
                Toast.makeText(RegestairActivity.this, "Netorwk error....", Toast.LENGTH_SHORT).show();

            }
        });


    }
}

File: ChatApp/app/src/main/java/com/example/chatapp/RegestairActivity.kt
package com.example.chatapp

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.FirebaseDatabase
import kotlinx.android.synthetic.main.activity_regestair.*

class RegestairActivity : AppCompatActivity() {

    private  lateinit var  faireauth:FirebaseAuth
    private  lateinit var databaseuser:DatabaseReference
    private var fairebaseUid:String?=null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_regestair)

        val toolbar:androidx.appcompat.widget.Toolbar=findViewById(R.id.toolbar_regestair)
        setSupportActionBar(toolbar)
        supportActionBar?.title="Register"
        supportActionBar?.setDisplayHomeAsUpEnabled(true);
        toolbar.setNavigationOnClickListener() {
            finish()
        }

        btn_regstair.setOnClickListener({
            register()








        })





    }

    private fun register() {
        faireauth= FirebaseAuth.getInstance()
        databaseuser=FirebaseDatabase.getInstance().getReference()
        var name:String=edit_name_regestair.text.toString();
        var email:String=edit_email_regstair.text.toString();
        var password:String=edit_password_regstair.text.toString();

        if(name==""){
            Toast.makeText(this,"Enter Name",Toast.LENGTH_SHORT).show()
        }else if(email==""){
            Toast.makeText(this,"Enter Email",Toast.LENGTH_SHORT).show()

        }else if(password==""){
            Toast.makeText(this,"Enter pass",Toast.LENGTH_SHORT).show()

        }else{
            faireauth.createUserWithEmailAndPassword(email,password).addOnCompleteListener {task->
                if (task.isSuccessful){
                    fairebaseUid=faireauth.currentUser!!.uid

                    var  map=HashMap<String,Any>()
                    map["uid"]=fairebaseUid.toString()
                    map["name"]=name
                    map["password"]=password.toString()
                    map["profile_photo"]="https://firebasestorage.googleapis.com/v0/b/secand-f1daf.appspot.com/o/download.png?alt=media&token=8b9e6106-f47d-40b9-ac87-fa1549d55318"
                    map["statue"]="offline"
                    map["search"]=name.toLowerCase()
                    map["facebook"]="https://m.facebook.com"
                    map["instagram"]="https://instagram.com"
                    map["website"]="https://wwww.google.com"

                    databaseuser.child("Users").child(fairebaseUid.toString()).updateChildren(map).addOnCompleteListener({task->
                       val intent= Intent(this,LoginActivity::class.java)
                       startActivity(intent)
                    })


                }else{
                    Toast.makeText(this,"Error"+task.exception.toString(),Toast.LENGTH_SHORT).show()

                }
            }

        }



    }
}